src/otaclient/ota_core/_updater_base.py

```diff
@@ -237,3 +274,154 @@ def _process_metadata(self, only_metadata_verification: bool = False) -> None:
         self._metrics.ota_image_total_symlinks_num = (
             self._ota_metadata.total_symlinks_num
         )
+
+
+DEFAULT_IMAGE_ID = ImageIdentifier(
+    ecu_id=ecu_info.ecu_id,
+    release_key=OTAReleaseKey.dev,
+)
+
+
+class OTAImageV1SupportMixin(OTAUpdateInterface):
```

Comment:
OTA image version specific implementation(mostly the metadata downloading and resources downloading) are implemented by the Mixin, which will be plugin to the detailed OTA implementation.